### PR TITLE
refactor(framework) Run exec servicer in SuperLink

### PR DESCRIPTION
**Pull request by panh99**

### CODE REVIEW

File: src/py/flwr/server/app.py [hunk 1]
```diff
@@ -681,3 +698,52 @@ def _add_args_fleet_api(parser: argparse.ArgumentParser) -> None:
...
+        help="The directory for the executor.",
+        default=".",
+    )
+    parser.add_argument(
+        "--executor-config",
+        help="Key-value pairs for the executor config, separated by spaces. "
+        "For example:\n\n`--executor-config 'verbose=true "
+        'root-certificates="certificates/superlink-ca.crt"\'`',
+    )
+
```
Comment (chongshenng) [inline src/py/flwr/server/app.py:727]: Do we need these other args for secure connection? 
```python
    parser.add_argument(
        "--executor-ssl-certfile",
        help="Exec API server SSL certificate file (as a path str) "
        "to create a secure connection.",
        type=str,
        default=None,
    )
    parser.add_argument(
        "--executor-ssl-keyfile",
        help="Exec API server SSL private key file (as a path str) "
        "to create a secure connection.",
        type=str,
    )
    parser.add_argument(
        "--executor-ssl-ca-certfile",
        help="Exec API server SSL CA certificate file (as a path str) "
        "to create a secure connection.",
        type=str,
    )    
```

**Review by chongshenng — DISMISSED** [review]
Thanks @panh99. I think we need the args for secure connection for the Exec API. Also, should we add warnings to the `flower-superexec` command? Something like:
```python
def run_superexec() -> None:
    """Run Flower SuperExec."""
    event(EventType.RUN_SUPEREXEC_ENTER)
    log(
        ERROR,
        "The command `flower-superexec` was deprecated as of version 1.12 "
        "and merged into `flower-superlink.",
    )
    log(INFO, "Execute `flower-superlink --help` to learn how to use it.")
    register_exit_handlers(event_type=EventType.RUN_SUPEREXEC_LEAVE)
```

File: src/py/flwr/server/app.py (hunk 1 above)
Comment (panh99) [reply src/py/flwr/server/app.py:727]: I don't think we need an extra tuple of certificates specifically for `Exec API`. These certificates allow the SuperLink to estabilish an SSL-enabled connection with a grpc client, and we can just reuse the certificates for `Fleet API` and `Driver API`.

**Comment by panh99** [comment]
> Thanks @panh99. I think we need the args for secure connection for the Exec API. Also, should we add warnings to the `flower-superexec` command? Something like:
> 
> ```python
> def run_superexec() -> None:
>     """Run Flower SuperExec."""
>     event(EventType.RUN_SUPEREXEC_ENTER)
>     log(
>         ERROR,
>         "The command `flower-superexec` was deprecated as of version 1.12 "
>         "and merged into `flower-superlink.",
>     )
>     log(INFO, "Execute `flower-superlink --help` to learn how to use it.")
>     register_exit_handlers(event_type=EventType.RUN_SUPEREXEC_LEAVE)
> ```

I think it's better to open a separate PR to add deprecation warnings.

File: src/py/flwr/server/app.py [hunk 2]
```diff
@@ -681,3 +698,52 @@ def _add_args_fleet_api(parser: argparse.ArgumentParser) -> None:
...
+
+def _load_executor(
```
Comment (danieljanes) [inline src/py/flwr/server/app.py:729]: @panh99 instead of copying this, can we make `_load_executor` (and the error class) public and use it here?

**Comment by chongshenng** [comment]
> @chongshenng I think it's better to open a separate PR to add deprecation warnings.
Good point, that would also be a breaking change.

File: src/py/flwr/common/constant.py [hunk 3]
```diff
@@ -40,15 +40,14 @@
...
-# SuperExec
-EXEC_API_DEFAULT_ADDRESS = "0.0.0.0:9093"
 # SuperLink
 DRIVER_API_DEFAULT_ADDRESS = "0.0.0.0:9091"
 FLEET_API_GRPC_RERE_DEFAULT_ADDRESS = "0.0.0.0:9092"
 FLEET_API_GRPC_BIDI_DEFAULT_ADDRESS = (
     "[::]:8080"  # IPv6 to keep start_server compatible
 )
 FLEET_API_REST_DEFAULT_ADDRESS = "0.0.0.0:9093"
+EXEC_API_DEFAULT_ADDRESS = "0.0.0.0:9095"
```
Comment (danieljanes) [inline src/py/flwr/common/constant.py:50]: @Robert-Steiner do we need to change anything else to make it work with the new default port number?

File: src/py/flwr/server/app.py (hunk 2 above)
Comment (panh99) [reply src/py/flwr/server/app.py]: Sure thing. I've updated the PR as per your suggestion.

File: src/py/flwr/common/constant.py (hunk 3 above)
Comment (panh99) [reply src/py/flwr/common/constant.py:50]: I changed the default port here b/c it conflicts with the default port for Rest-based Fleet API. We can also updated the default port for that one and keep the default port for exec API unchanged if it's easier to do so.

Comment (Robert-Steiner) [reply src/py/flwr/common/constant.py:50]: I'm fine with `9095`. We need to update the Docker docs anyway as we need to remove SuperExec from it.

Comment (danieljanes) [reply src/py/flwr/common/constant.py:50]: @panh99 great idea, let's change the default port for the REST API (which is not commonly used at this point)

**Review by danieljanes — APPROVED** [review]

**Review by danieljanes — APPROVED** [review]

**Review by danieljanes — APPROVED** [review]

**Pull request merged**